.github/actions/setup/action.yml

```diff
@@ -0,0 +1,29 @@
+name: 'Setup Android Build Environment'
+description: 'Sets up JDK and Google Services configuration for Android builds'
+
+inputs:
+  google-services-json:
+    description: 'Content of google-services.json file'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    # Sets up a JDK on the job so that we can run Java code
+    - name: Set up JDK 17
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'zulu'
+        java-version: '17'
+        java-package: jdk
+
+    - name: Add Google Services from Secrets
```

Comment:
The main reason I got into looking at the workflows -- inserting the google services file is required for all these CI runs that use gradle.